tools/lambda-promtail/lambda-promtail/s3.go

```diff
@@ -131,11 +131,14 @@ func parseS3Log(ctx context.Context, b *batch, labels map[string]string, obj io.
 		}
 
 		match := timestampRegex.FindStringSubmatch(log_line)
+		var timestamp time.Time
 		if len(match) > 0 {
 			timestamp, err = time.Parse(time.RFC3339, match[1])
 			if err != nil {
 				return err
 			}
+		} else {
+			timestamp = time.Now()
```

Comment:
we could simplify further by just having `timestamp = time.Now()` as the first line of the loop, and declare `var timestamp time.Time` immediately before the loop

removes the need for an `else`